File: RecipeApp/Models/SavedRecipe.cs
namespace RecipeApp.Models;

public class SavedRecipe
{
    public int Id { get; set; }
    public string UserId { get; set; } = string.Empty;
    public string Title { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public string SkillLevel { get; set; } = "Beginner";
    public int CookTimeMinutes { get; set; } = 30;
    public string IngredientsJson { get; set; } = "[]";
    public string Instructions { get; set; } = string.Empty;
    public string? SuggestedAdditionsJson { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
}

File: server/src/models/SavedRecipe.ts
export interface SavedRecipe {
  id: number;
  userId: string;
  title: string;
  description: string;
  skillLevel: string;
  cookTimeMinutes: number;
  ingredientsJson: string;
  instructions: string;
  suggestedAdditionsJson?: string;
  createdAt: Date;
}

// Helper type for API responses (parsed JSON fields)
export interface SavedRecipeDto {
  id: number;
  userId: string;
  title: string;
  description: string;
  skillLevel: string;
  cookTimeMinutes: number;
  ingredients: string[];
  instructions: string[];
  suggestedAdditions: string[];
  createdAt: Date;
}
